serving/purgatorio/select-and-train-machine-learning-models/deep-learning-theory/deep-learning-theory.md

```diff
@@ -0,0 +1,324 @@
+
+Some useful papers in understanding Deep Learning theory are listed at the end of this guide.
+
+Does this mean that you won't be able to train a neural net well? Not really! In fact, you'll probably get better results than you could ever get with any other method, and the hard thing will be understanding **why you get them!** 
+
+With experience, you'll develop an insight into which approaches work well and which don't, how to fix existing architectures to your specific problems, and so on.
+
+You already know that training a Neural Network involves using a training dataset to update the model weights to create a good mapping of inputs to outputs.
+
+This training process is solved using an optimization algorithm that searches through a space of possible values for the neural network model weights for a set of weights that results in a good performance on the training dataset.
+
+[Training Neural Networks can be hard](https://machinelearningmastery.com/why-training-a-neural-network-is-hard/), and some of the reasons are listed here:
+
+[37 Reasons why your Neural Network is not working](https://blog.slavv.com/37-reasons-why-your-neural-network-is-not-working-4020854bd607)
+
+Of course, your small CNN is performing well when testing it on a benchmark dataset like MNIST, but how does it work against a real-life dataset, maybe with millions of images? 
+
+Training Neural Networks effectively requires experience and even the most expert researchers try a lot of different neural architectures before the results they want. 
+
+It's hard to find a list on the internet that includes expert knowledge and best practices, and that's what Virgilio tries to do in this section.
+
+First things first:
+
+It's **strongly recommended** to refer to [this page](http://cs231n.github.io/) from Stanford and go through all the Modules 1 and 2.
+
+
+### Understanding Backpropagation
+
+Backpropagation is the mathematical "magical trick" that powers the training of neural networks. Basically, it allows using the Gradient Descent iterative algorithm to find the optimal weights of the network (the connections among the neurons).
+
+Understanding backpropagation is simple if you have the necessary math basics (you can find them in the [Math Fundamentals](serving\purgatorio\fundamentals\math-fundamentals\math-fundamentals.md) Guide).
+
+The basic reference you need to understand backpropagation is the following chapter of the book [Neural Networks and Deep Learning](http://neuralnetworksanddeeplearning.com/):
+
+[**How the backpropagation works**](http://neuralnetworksanddeeplearning.com/chap2.html)
+
+Once you're done with this reading, you can play with some other different explanations in order to deepen your understanding (remember that visualization is awesome for learning, as stated in the [Virgilio's Teaching Strategy](serving\paradiso\virgilio-teaching-strategy\virgilio-teaching-strategy.md) Guide.
```

Comment:
I think a video or animation on what or how backpropagation works would help at this stage.

Reply:
there is already from 3blue1brown

Reply:
Where? If so, please link it here somewhere and ask them to look at it to clarify their understanding and then move on!